### ISSUE TITLE
Snapjaw mobs in Alterac Mountains pathing/evading issues

### ISSUE THREAD

**Issue by CosmicCleric** [chromiecraft/chromiecraft#3954]
### What client do you play on?

enUS

### Faction

Alliance

### Content Phase:

30-39

### Current Behaviour

Snapjaw mobs that are in the river that runs North/South through Alterac Mountains (and Hillsbrad Foothills) have pathing issues. River starts East of Southshore 55.0 62.1 and goes North from there.

When you commence combat with them, it causes them to evade/retreat/reset after being in combat for a few seconds. Auto attacks will not work, but you can use spells to attack and kill them, until they evade/reset.

Sometimes they will swim up/down the river for a while before attempting to come onto land and come to you, to start combat. This seems to happen more often in narrow passages of the river. This does not happen all the time.

If you start combat in water, they will pop up/down vertically as if they're trying to decide if they are swimming or walking on top of the river water.

The murlocs West of Southshore have the same behavior issues pathing wise (but not evading/resetting).

### Expected Blizzlike Behaviour

Be able to enter combat where you can hit the mobs with autoattacks, and the mobs confront you directly and not run away, cross a point, then run all the way back to you. Also not evading/resetting after being in combat for a while and their health is 50% or lower.

### Source

Retail/Classic WoW experiences of how mobs should act when attacked.

### Steps to reproduce the problem

1. Stand on the bank of the river and target a nearby Snapjaw.
2. Commence combat (I use the Exorcism spell).
3a. They may swim away from you up/down the river for a while, then come to shore, then run down the showline towards you.
3b. They may just face you and start combat.
4. You'll notice that you cannot autoattack the Snapjaws, but can cast spells on them.
5. When combat has been ongling for fiveish seconds, the Snapjaw may show 'Evade' and return to their starting spot, in the river.

### Extra Notes

As I mentioned previous, the murlocs west Southshore more rarely but do demonstrate the same pathing out of water to land problem, but not the evade problem.

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/b2e449fd15e38b4778a42650f388b8213cf62a6a

### Operating system

Kubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-bg-item-reward](https://github.com/azerothcore/mod-bg-item-reward)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-cta-switch](https://github.com/azerothcore/mod-cta-switch)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-dmf-switch](https://github.com/azerothcore/mod-dmf-switch)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna](https://github.com/azerothcore/mod-eluna)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-low-level-rbg](https://github.com/azerothcore/mod-low-level-rbg)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-progression-system](https://github.com/azerothcore/mod-progression-system)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-rdf-expansion](https://github.com/azerothcore/mod-rdf-expansion)
- [mod-transmog](https://github.com/azerothcore/mod-transmog)
- [mod-weekend-xp](https://github.com/azerothcore/mod-weekend-xp)
- [mod-instanced-worldbosses](https://github.com/nyeriah/mod-instanced-worldbosses)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-exchange-npc](https://github.com/55Honey/Acore_ExchangeNpc)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### Customizations

None

### Server

ChromieCraft

**Comment by BeefSteaker** [comment]
Hi, thanks for the report. 
i have tried attacking these mobs multiple locations along the river, and had no issues with evading

its been a while since you made this report, it may have been fixed by something else along the way

**Comment by CosmicCleric** [comment]
> Hi, thanks for the report. i have tried attacking these mobs multiple locations along the river, and had no issues with evading
> 
> its been a while since you made this report, it may have been fixed by something else along the way

You had to go up-river quite a way to find the area where the pathing issues occurred. Where did you test at (on the river)?

Was there a major fix for mob pathing issues? If so, then it may have fixed this as well.

I'd hate to have this closed if its still a problem, really detracts from the server (the bug).

**Comment by BeefSteaker** [comment]
Definitely been a couple tweaks to water pathing specifically yeah. I'll have a double check tomorrow to make sure. 

If you can remember some exact spots and check yourself too that'd be great

**Comment by CosmicCleric** [comment]
> Definitely been a couple tweaks to water pathing specifically yeah. I'll have a double check tomorrow to make sure.
> 
> If you can remember some exact spots and check yourself too that'd be great

Hey, so I just checked it, and I'm seeing the problem less often, but I'm still seeing the problem.

Happened several times, MY (not mob) locations: 

- 68.14 20.60
- NE of Tarren Mill
- 89.75 47.04/90.23 46.67 (same mob I started attack two locations).

I'm swimming in the middle of the river and using Exorcism to start combat with a mob at range.

There's definitely an improvement, but still an issue. Should reopen the ticket.
